client/src/client/pm.ts

```diff
@@ -0,0 +1,141 @@
+const Hawk = require('@hapi/hawk/lib/browser');
+import { PaymentsSearchFields } from '../PaymentsType';
+import { parseObjToQueryString } from '../utils/parseObjToQueryString';
+
+function getPMUrl(): string | undefined {
+  const localPmUrl = window.localStorage.getItem('pm_url');
+  const { PM_URL } = process.env;
+
+  if (localPmUrl && localPmUrl.length > 0) {
+    return localPmUrl;
+  } else if (PM_URL && PM_URL !== '') {
+    return PM_URL;
+  } else if (location.hostname === 'localhost' || location.hostname === '127.0.0.1') {
+    return;
+  }
+
+  return window.location.origin.replace('console', 'payment');
+}
+
+async function request(
+  method: 'GET' | 'POST' | 'DELETE' | 'PUT',
+  endpoint: string,
+  params?: any,
+): Promise<Response> {
+  const headers = new Headers();
+  const pmUrl = getPMUrl();
+  const requestUrl = `${pmUrl}${endpoint}`;
+
+  if (!pmUrl) {
+    console.error('PaymentManager URL missing');
+    return Promise.reject();
```

Comment:
```suggestion
    return Promise.reject('PaymentManager URL missing');
```
Perhaps this way will be better?